tests/test.py

```diff
@@ -0,0 +1,24 @@
+import unittest
+from unittest import mock
+import buddy_config
+
+class TestTypes(unittest.TestCase):
+
+    @mock.patch.dict("os.environ", {"BOOL_VALUE": "tRue"})
+    def test_boolean_conversion(self):
+        class MyConf(metaclass=buddy_config.Config):
+            BOOL_VALUE = "BOOL_VALUE", buddy_config.Boolean
```

Comment:
This is a bit of of overhead for the user, you need to know the libraries' custom types. Perhaps you can add above an `isinstance(var, bool)` then return the `buddy_config.Boolean` class?

Reply:
The problem that `bool` is not a class it is a single per-process link to a C level variable. They are two 1 and 0. `isinstance` doesn't work with bool. So `isinstance(True, int) == True` and this is not helpful because it pushes us to a more tricky operation with a determination which int is actually `bool`.

Reply:
I'll try it. Thanks.

Reply:
Yep, it is cleaner. Thank you! 

I mixed several concepts and overcomplicated it. Now it is pretty simple, just a wrapper around a regex validation.